daemon/graphdriver/overlay2/check.go

```diff
@@ -107,3 +107,59 @@ func doesSupportNativeDiff(d string) error {
 
 	return nil
 }
+
+// doesMetacopy checks if the filesystem is going to optimize changes to
+// metadata by using nodes marked with an "overlay.metacopy" attribute to avoid
+// copying up a file from a lower layer unless/until its contents are being
+// modified
+func doesMetacopy(d string) (bool, error) {
```

Comment:
The important thing to note here is that we are testing if metacopy is being used -- we (the daemon) don't actually ask for it to be enabled, or not enabled. That is left entirely up to the host OS (which may have it set due to `CONFIG_OVERLAY_FS_METACOPY` or `overlay2.metacopy`).

This is meant to indicate if metacopy is being used, not if it is supported. Obviously if it is unsupported it will not be used.

Reply:
> The important thing to note here is that we are testing if metacopy is being used

Ah, gotcha; I should've looked more in-depth, yes in that case naming makes sense